modules/livestore/instance.go

```diff
@@ -29,7 +30,7 @@ type instance struct {
 	enc encoding.VersionedEncoding
 
 	// Block management
-	blocksMtx      sync.Mutex
+	blocksMtx      sync.RWMutex
```

Comment:
Needed more fine grained mutex when dealing with blocks.

Reply:
rwmutex's are slower than plain ol' mutexes. unless it's heavy on reads a lot of times a plain mutex is faster. no idea what the ratio is.

Reply:
The ingester code used RWMutex so assuming we thought about that then :)